src/pages/task/taskTabs/buildBaronAndAnnotations/AddIssueModal/state.ts

```diff
@@ -0,0 +1,95 @@
+import { useReducer } from "react";
+import { useSpruceConfig } from "hooks";
+import { numbers } from "utils";
+import { validateJiraURL } from "utils/validators";
+
+const { toDecimal } = numbers;
+
+interface AddIssueState {
+  url: string;
+  issueKey: string;
+  canSubmit: boolean;
+  isURLValid: boolean;
+  isKeyValid: boolean;
+  confidenceScore: string;
+}
+
+type Action =
+  | { type: "reset" }
+  | { type: "setUrl"; url: string; jiraURL: string }
+  | { type: "setKey"; issueKey: string }
+  | { type: "setConfidenceScore"; confidenceScore: string };
+
+const init = (): AddIssueState => ({
+  url: "",
+  issueKey: "",
+  canSubmit: false,
+  isURLValid: false,
+  isKeyValid: false,
+  confidenceScore: null,
+});
+
+const reducer = (state: AddIssueState, action: Action) => {
+  switch (action.type) {
+    case "reset":
+      return init();
+    case "setUrl": {
+      const isURLValid = validateJiraURL(action.jiraURL, action.url);
```

Comment:
![image](https://user-images.githubusercontent.com/13104717/167197356-19e34ea7-5dc4-409b-acb1-6ec95891abb5.png)
This doesn't provide feedback to the user as to why saving is blocked which may confuse users.

Reply:
![image](https://user-images.githubusercontent.com/13104717/167197583-f9f937c8-22c1-44a9-a0a9-9a9de9d4aa68.png)
the link should not have to be a Jira link. we should only perform this validation if it's a Jira ticket url.

Reply:
And if we are adding validation, ideally, if it's a Jira ticket, the display text should be pulled from Jira and we should disable the Display Text field so that we don't end up with something like this: 
![image](https://user-images.githubusercontent.com/13104717/167199036-28448f75-f058-494b-92ea-1505d5839fe3.png)


This may be out of scope for this PR though.

Reply:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/13104717/169356809-7ea8f8ac-4aa5-44f4-bbe6-bb1b46a05fae.png">
Are we sure we want this error in a toast as opposed to inline in the form?

Reply:
It's a matter of preference, I'm fine with either.

Reply:
Resolved it. I blocked submission on invalid urls